plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StagingDirectoryProvider.java

```diff
@@ -0,0 +1,18 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.core.resources.IProject;
+
+/**
+ * Common interface class to obtain the staging directory for App Engine Standard deploy.
+ */
+public interface StagingDirectoryProvider {
+
+  /**
+   * @param event the event that was triggered to run the App Engine Standard deploy
+   * @return path to the directory to be used for staging
+   * @see com.google.cloud.tools.eclipse.appengine.deploy.standard.ProjectToStagingExporter#writeProjectToStageDir(IProject, String)
+   * @see com.google.cloud.tools.eclipse.appengine.deploy.standard.StandardDeployCommandHandler
+   */
+  String getStagingDirectory();
```

Comment:
If this is truly a library class, perhaps `getDirectory()` or just `get()` (like Java8's `java.util.function.Supplier`).